File: WEBSITE_FF_NEW_V4_CU/minify.py
import re
import os

def minify_css(css_content):
    """Minify CSS by removing comments, whitespace, and newlines"""
    # Remove comments
    css_content = re.sub(r'/\*[\s\S]*?\*/', '', css_content)
    # Remove excess whitespace
    css_content = re.sub(r'\s+', ' ', css_content)
    # Remove spaces around {, }, :, ;, ,
    css_content = re.sub(r'\s*([{}:;,])\s*', r'\1', css_content)
    return css_content.strip()

def minify_js(js_content):
    """Minify JS by removing comments, whitespace, and newlines"""
    # Remove multi-line comments
    js_content = re.sub(r'/\*[\s\S]*?\*/', '', js_content)
    # Remove single-line comments (but preserve URLs)
    js_content = re.sub(r'([^:]|^)//.*$', r'\1', js_content, flags=re.MULTILINE)
    # Remove excess whitespace
    js_content = re.sub(r'\s+', ' ', js_content)
    # Remove spaces around operators
    js_content = re.sub(r'\s*([{}();,:])\s*', r'\1', js_content)
    return js_content.strip()

files_to_minify = [
    ('assets/css/style.css', 'assets/css/style.min.css', 'css'),
    ('assets/css/chat_style.css', 'assets/css/chat_style.min.css', 'css'),
    ('assets/js/script.js', 'assets/js/script.min.js', 'js'),
    ('assets/js/chat_bot.js', 'assets/js/chat_bot.min.js', 'js'),
]

print('🔧 Minifying files...\n')

for input_file, output_file, file_type in files_to_minify:
    try:
        with open(input_file, 'r', encoding='utf-8') as f:
            content = f.read()
        
        minified = minify_css(content) if file_type == 'css' else minify_js(content)
        
        with open(output_file, 'w', encoding='utf-8') as f:
            f.write(minified)
        
        original_size = len(content.encode('utf-8')) / 1024
        minified_size = len(minified.encode('utf-8')) / 1024
        savings = ((len(content) - len(minified)) / len(content)) * 100
        
        print(f'✅ {input_file}')
        print(f'   → {output_file}')
        print(f'   Size: {original_size:.2f}KB → {minified_size:.2f}KB ({savings:.1f}% smaller)\n')
    except Exception as e:
        print(f'❌ Error: {input_file} - {str(e)}\n')

print('✨ Minification complete!')

File: WEBSITE_FF_NEW_V4_CU/minify.js
const fs = require('fs');
const path = require('path');

// Simple JS Minifier (removes comments, whitespace, newlines)
function minifyJS(code) {
    return code
        // Remove multi-line comments
        .replace(/\/\*[\s\S]*?\*\//g, '')
        // Remove single-line comments (but preserve URLs)
        .replace(/([^:]|^)\/\/.*$/gm, '$1')
        // Remove excess whitespace
        .replace(/\s+/g, ' ')
        // Remove spaces around operators and punctuation
        .replace(/\s*([{}();,:])\s*/g, '$1')
        // Remove newlines
        .replace(/\n/g, '')
        .trim();
}

// Simple CSS Minifier
function minifyCSS(code) {
    return code
        // Remove comments
        .replace(/\/\*[\s\S]*?\*\//g, '')
        // Remove excess whitespace
        .replace(/\s+/g, ' ')
        // Remove spaces around {, }, :, ;, ,
        .replace(/\s*([{}:;,])\s*/g, '$1')
        // Remove newlines
        .replace(/\n/g, '')
        .trim();
}

// Process files
const files = [
    { input: 'assets/css/style.css', output: 'assets/css/style.min.css', type: 'css' },
    { input: 'assets/css/chat_style.css', output: 'assets/css/chat_style.min.css', type: 'css' },
    { input: 'assets/js/script.js', output: 'assets/js/script.min.js', type: 'js' },
    { input: 'assets/js/chat_bot.js', output: 'assets/js/chat_bot.min.js', type: 'js' }
];

console.log('🔧 Minifying files...\n');

files.forEach(file => {
    try {
        const input = fs.readFileSync(file.input, 'utf8');
        const minified = file.type === 'css' ? minifyCSS(input) : minifyJS(input);

        fs.writeFileSync(file.output, minified);

        const originalSize = (Buffer.byteLength(input) / 1024).toFixed(2);
        const minifiedSize = (Buffer.byteLength(minified) / 1024).toFixed(2);
        const savings = (((Buffer.byteLength(input) - Buffer.byteLength(minified)) / Buffer.byteLength(input)) * 100).toFixed(1);

        console.log(`✅ ${file.input}`);
        console.log(`   → ${file.output}`);
        console.log(`   Size: ${originalSize}KB → ${minifiedSize}KB (${savings}% smaller)\n`);
    } catch (error) {
        console.error(`❌ Error processing ${file.input}:`, error.message);
    }
});

console.log('✨ Minification complete!');
